core_impl/src/test/java/com/google/instrumentation/stats/StatsManagerImplTest.java

```diff
@@ -192,16 +199,26 @@ public void testRecordNonExistentMeasurementDescriptor() {
     assertThat(view.getDistributionAggregations()).hasSize(0);
   }
 
-  // TODO(songya): update this test once we determine how to handle tags that aren't an exact match
-  // for the view.
+
   @Test
   public void testRecordNonExistentTag() {
     statsManager.registerView(RpcConstants.RPC_CLIENT_ROUNDTRIP_LATENCY_VIEW);
-    statsManager.record(
-        wrongTag, MeasurementMap.of(RpcConstants.RPC_CLIENT_ROUNDTRIP_LATENCY, 10.0));
+    statsManager.record(wrongTag, MeasurementMap.of(
+        RpcConstants.RPC_CLIENT_ROUNDTRIP_LATENCY, 10.0));
+    statsManager.record(wrongTag, MeasurementMap.of(
+        RpcConstants.RPC_CLIENT_ROUNDTRIP_LATENCY, 50.0));
```

Comment:
I think it would be useful to check that stats for multiple StatsContexts with non-matching keys are combined.  You could use a second set of tags here or create a new test.